src/stream.ts

```diff
@@ -151,6 +152,15 @@ class StreamState {
   }
 }
 
+// Max message size that can be sent to the DataChannel
+const MAX_MESSAGE_SIZE = 16 * 1024
+
+// How much can be buffered to the DataChannel at once
+const MAX_BUFFERED_AMOUNT = 16 * 1024 * 1024
+
+// How long time we wait for the 'bufferedamountlow' event to be emitted
+const BUFFERED_AMOUNT_LOW_TIMEOUT = 30 * 1000
```

Comment:
These are fine as defaults (I think) but the the user needs to be able to override them.

When creating the transport they should be able to pass these in as options:

```js
const node = await createLibp2p({
  transports: [
    webRTC({
      maxDataChannelMessageSize: number,
      maxDataChannelBufferedAmount: number,
      dataChannelBufferAmountLowEventTimeout: number
    })
     //.. other transports
  ]
  //.. other config
})
```

They should be added to the [WebRTCTransportInit](https://github.com/libp2p/js-libp2p-webrtc/blob/main/src/private-to-private/transport.ts#L24-L26) interface with some appropriate JSDoc comments explaining what they do so users get nice IntelliSense-style help prompts when using them.

Reply:
Add added the args as 

```typescript
 webRTC({
      constraints: {
          maxDataChannelMessageSize: number,
          maxDataChannelBufferedAmount: number,
          dataChannelBufferAmountLowEventTimeout: number
      }
    })
```

else moving the options around not that nice... 

More minimalistic woiuld be
```typescript
 webRTC({
      ...other,
      datachannel: {
          maxMessageSize: number,
          maxBufferedAmount: number,
          timeout: number
      }
    })
```

?


Also it kind of also make sense to add a test to see whether the arguments are propagating through correctly, both when you initiate a connection, and also when you are on the receiving end.

Reply:
If having everything as a top-level option is beyond the pale, I think this is preferable:

```js
 webRTC({
      ...other,
      dataChannel: {
          maxMessageSize: number,
          maxBufferedAmount: number,
          timeout: number
      }
    })
```

This way it's clear the limits apply to the data channel, but `timeout` needs something to disambiguate it - my intuition would be that it's a stream inactivity timeout but that's not the case here.

Reply:
Soo this is a ```timeout``` for unresponsiveness. But generally ```timeout``` in connections means that no party has sent any message for a while, so we close the connection. So perhaps this makes sense
```typescript
 webRTC({
      ...other,
      
      dataChannel: {
          maxMessageSize: number,
          maxBufferedAmount: number,
          bufferTimeout: number,
          keepAliveTimeout: number, // not a feature now, but in theory
      }
    })

```